docs/developer-guide/LayerFilter.md

```diff
@@ -0,0 +1,324 @@
+}
+```
+
+By default all the filters contained in the "filters" array are combined with an `AND` operator. Every filter in the array is a JSON object that can be serialized in different formats (CQL, OGC, etc.).
+Each of them is a JSON object that must have the `format` value, to be recognized and properly converted,
+
+All the filters in `filters` array will have at least a `format` field and an optional `id` attribute reserved, that can be used to identify the filter from a component that wants to use it.
+
+Filters in the `filters` array can be combined with a logic operator (`AND`, `OR`), in this case the filter object must have the `logic` format.
+Moreover they can be of `mapstore` format too.
+
+!!! Note
+    For backward compatibility, the filter object without the "format" field is considered as "mapstore" format, version "1.0.0".
+
+!!! Note
+    For backward compatibility, the filters "mapstore" of version "1.0.0" can contain also several other fields that will be deprecated in the future in favor of a `mapstore-query-panel` format, that is the format used by the query panel, and currently mostly supported in MapStore.
+    So a filter like this is still valid:
+
+        ```json
+        {
+            "format": "mapstore",
+            "version": "1.0.0",
+            "groupFields": [],
+            "spatialField": {
+                "method": "BBOX",
+                "attribute": "the_geom",
+                "operation": "INTERSECTS",
+                "geometry": {
+                    "type": "Polygon",
+                    "coordinates": [[[12.5,41.5], [12.5,  42.5], [13.5,42.5], [13.5, 41.5], [12.5,41.5]]]
+                }
+            },
+            "filterFields": [{
+                "attribute": "name",
+                "operator": "=",
+                "type": "string",
+                "value": "test"
+            }]
+        }
+        ```
+
+### `logic` format
+
+In order to allow to create complex filters, MapStore allows to combine filters with a logic operator (`AND`, `OR`).
+The `logic` format is a JSON object that has this shape:
+
+```json
+{
+    "format": "logic",
+    "version": "1.0.0",
+    "logic": "AND",
+    "filters": []
+}
+```
+
+### `cql` format
+
+The `cql` format is a JSON object that has this shape:
+
+```json
+{
+    "format": "cql",
+    "version": "1.0.0",
+    "body": "..."
+}
+```
+
+!!! Note:
+    MapStore actually supports only a subset of CQL, that is the one used by GeoServer. In particular can not parse WKT geometries yet.
+
+### `mapstore-query-panel` format
+
+The `mapstore-query-panel` format is a JSON object that has this shape:
+
+```json
+{
+    "format": "mapstore-query-panel",
+    "version": "1.0.0",
+    "groupFields": [],
+    "spatialField": {},
+    "filterFields": [],
+    "crossLayerFilter": {},
+
+}
+```
+
+Now it do not have an implementation yet, but this format will replace the old legacy 'mapstore' fields in the future.
+
+## Supporting new formats
```

Comment:
I should not add registerConverter for the moment because the API may change, so I should keep the essential.
Going to fix the documentation to clarify these points.